From 07cc2a26a4594f02370697b2644350520c0c2ed3 Mon Sep 17 00:00:00 2001
From: Charles Plessy <plessy@debian.org>
Date: Sun, 26 Feb 2012 18:24:29 +0900
Subject: [PATCH] Relocation of the upgrading checklist.

 - Move the hypertext upgrading checklist to the policy.html directory.
 - Stop building and distributing the multi-page HTML version.
 - Stop distributing the SGML version in the binary package.
 - The location of the compressed text version does not change.

Note that the upgrading checklist must be built before policy.html.tar.gz,
otherwise it will not be included in the binary package.

Closes: #639663
---
 Makefile     |    4 ++++
 debian/rules |   12 ++++++++----
 2 files changed, 12 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 1f3675c..18d88b9 100644
--- a/Makefile
+++ b/Makefile
@@ -24,6 +24,10 @@ perl-policy.sgml: version.ent
         mv $*-1d.html/index.html $*-1.html && \
         rmdir $*-1d.html
 
+policy.html/upgrading-checklist.html: upgrading-checklist.sgml
+	LANG=C debiandoc2html -1 -b policy/upgrading-checklist upgrading-checklist.sgml && \
+	mv policy.html/upgrading-checklist-index.html policy.html/upgrading-checklist.html
+
 %.html.tar.gz: %.html/index.html
 	tar -czf $(<:/index.html=.tar.gz) $(<:/index.html=)
 
diff --git a/debian/rules b/debian/rules
index d5d93cf..f2130a8 100755
--- a/debian/rules
+++ b/debian/rules
@@ -32,7 +32,7 @@ LIBDIR	  := $(TMPTOP)/usr/share/doc-base
 
 sanitycheck := debian/rules policy.sgml
 
-SGML_FILES := policy menu-policy perl-policy upgrading-checklist
+SGML_FILES := policy menu-policy perl-policy
 DESC_FILES := copyright-format-1.0 debian-policy debian-menu-policy \
 	      debian-perl-policy debconf-spec fhs
 
@@ -54,7 +54,8 @@ POLICY_FILES = $(SGML_FILES:=.sgml) $(SGML_FILES:=.txt.gz) \
                debconf_spec/debconf_specification.html \
                debconf_spec/debconf_specification.txt.gz \
                policy.ps.gz policy.pdf.gz README.txt README.html \
-               Process.txt Process.html
+               Process.txt Process.html \
+               upgrading-checklist.txt.gz
 
 FILES_FROM_ORG := Process.html Process.txt README.txt README.html
 
@@ -69,7 +70,8 @@ FILES_TO_CLEAN  = copyright-format-1.0.xml.tar.gz \
 		  debconf_specification.xml.tar.gz \
 		  policy.pdf policy.ps policy.txt policy. \
 		  body.tmp head.tmp policy.tpt \
-		  $(FILES_FROM_ORG)
+		  $(FILES_FROM_ORG) \
+		  upgrading-checklist.txt.gz
 
 STAMPS_TO_CLEAN := stamp-policy stamp-build
 DIRS_TO_CLEAN   := debian/tmp fhs $(SGML_FILES:=.html)
@@ -83,7 +85,9 @@ make_directory  := install -p -d	-o root -g root	 -m  755
 all build build-indep: stamp-build
 build-arch:
 stamp-build: version.ent $(sanitycheck)
-	$(MAKE) $(SGML_FILES:=.sgml.validate) \
+	$(MAKE) policy.html/upgrading-checklist.html \
+		upgrading-checklist.txt.gz \
+		$(SGML_FILES:=.sgml.validate) \
 		$(SGML_FILES:=.html.tar.gz) \
                 $(SGML_FILES:=-1.html) \
 		$(SGML_FILES:=.txt.gz) \
-- 
1.7.9

